sdk/core/azure-core/inc/azure/core/internal/http/user_agent.hpp

```diff
@@ -11,12 +11,25 @@
 #include <string>
 
 namespace Azure { namespace Core { namespace Http { namespace _detail {
+  // NOTE: Treat Azure::Core::Http::_detail::UserAgentGenerator::GenerateUserAgent() as _internal -
+  // it is being/has been used by eventhubs.
   class UserAgentGenerator {
   public:
     static std::string GenerateUserAgent(
         std::string const& componentName,
         std::string const& componentVersion,
         std::string const& applicationId,
         long cplusplusValue);
+
+    [[deprecated("Use an overload with additional cplusplusValue parameter.")]] static std::string
```

Comment:
This will break EventHubs without a corresponding EH patch because code compiled with -WX will fail.

This *might* be ok in vcpkg scenarios but it will break non vcpkg scenarios.

Reply:
Given this is in details (and an implementation detail), I suggest we remove the deprecated label.

We can adjust our implementation post release, especially since this is a stop gap patch.

Reply:
The previously released eventhubs will compile in vcpkg, but should we within the SDK repo use that signature, it will emit deprecation warning, which will be treated as error.

Reply:
What about any customer who uses the SDK outside of vcpkg? I don't think there's value for us to see such a warning (as SDK authors/contributors), if we risk such a warning leaking to customers and breaking them on upgrade.

To understand the extent, would anyone using existing GA'd packages (say keyvault, storage, identity), see the deprecation warning, if they acquire the packages outside of vcpkg?

Reply:
We have done this exact pattern several times in the past, and if we care about it, than we perhaps should not be using deprecated all together.
What could happen is 1. new release of event hubs that does not use the deprecated functionality 2. when being consumed outside of vcpkg, SDK should not be built with warnings treated as errors.

Reply:
We have? I haven't seen any.

The deprecated attribute makes sense on **public** surface area (to let **end users** know), but it doesn't make sense for code that's internal or private to the SDK, that we can fix.

Reply:
> We have? I haven't seen any.

There are other `[[deprected`s, but this one I know about -
https://github.com/Azure/azure-sdk-for-cpp/blob/54e5e22b1a48fd9035f6d4421872a4050bea99f4/sdk/core/azure-core/inc/azure/core/credentials/credentials.hpp#L115

In this case, should you have old Identity and new Core, and fetch the sources, and build them with WARNINGS_AS_ERRORS, it will also explode.

Reply:
The difference there is that it is part of the public surface area that end users call/modify. If any end user inherited from a public class like TokenCredential, and overrode the protected method, it is incumbent on us to notify them with the deprecated attribute.

That doesn't apply for private/_detail methods and types.

Reply:
Yes, but it is also being used by Identity credentials, so many users are not using it directly, yet their `ClientSecretCredential`s will explode.
I am working on another PR, to disable treating warnings as errors when fetcing via FolderList.

Reply:
#6066